pkg/feeder/feeder.go

```diff
@@ -52,15 +53,17 @@ func NewClient(baseURL, baseAPI string, client *HttpClient) *Client {
 	}
 
 	// retry mechanism for do requests
+	// https://cloud.google.com/iot/docs/how-tos/exponential-backoff
 	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient) (*http.Response, error) {
 		var res *http.Response
-		wait := 2 * time.Second
+		wait := 1 * time.Second
+		const maxWait = 10 * time.Minute
 		for {
 			res, err = httpClient.Do(req)
 			if err != nil || res == nil || res.StatusCode != http.StatusOK {
-				wait *= 2
 				Logger.With("Waiting:", wait.Seconds(), "Error", err).Info("Waiting to do again a request")
 				time.Sleep(wait)
+				wait = time.Duration(math.Min((wait * 2).Seconds(), maxWait.Minutes()))
```

Comment:
```suggestion
				if newWait := wait * 2; newWait < maxWait {
					wait = newWait
				}
```
The `time.Duration` type is an `int64`, which means you can do the arithmetic operations directly. Also, I think an `if` statement is more clear than using `math.Min` and cast `time.Duration` to `float64`. Another detail is that you don't need to use the `Seconds()` or `Minutes()` functions because then you are overwriting the value of the current `wait` with a new `int64` that represents seconds or minutes and is used as nanoseconds by `time.Sleep()`.